app/controllers/application_controller.rb

```diff
@@ -29,13 +30,23 @@ def add_pagination_headers(collection)
 
   private
 
-  def authenticate_from_token
-    token = AuthToken.find_authenticated(
-      {
-        secret: request.headers[:'x-secret'] || params[:secret],
-        device_id: request.headers[:'x-device-id'] || params[:device_id]
-      }
-    )
+  def authenticate_user
+    token = authenticate_with_http_token { |it| AuthToken.find_by_token_for(:api, it) }
+    token ||= AuthToken.find_by_token_for(:api, params[:token])
```

Comment:
This is for things like audio requests, I assume?

Reply:
Yes, streaming audio is probably the only case where we cannot (always) use headers